modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
-            proxy.stop();
-            throw e;
-        }
+    public CouchbaseContainer withEnabledServices(final CouchbaseService... enabled) {
+        checkNotRunning();
+        this.enabledServices = EnumSet.copyOf(Arrays.asList(enabled));
+        return this;
     }
 
-    @SneakyThrows
-    private void startProxy(String networkAlias) {
-        proxy = new SocatContainer().withNetwork(getNetwork());
-
-        for (CouchbasePort port : CouchbasePort.values()) {
-            if (port.isDynamic()) {
-                proxy.withTarget(port.getOriginalPort(), networkAlias);
-            } else {
-                proxy.addExposedPort(port.getOriginalPort());
-            }
-        }
-
-        proxy.setWaitStrategy(null);
-        proxy.start();
-
-        ExecCreateCmdResponse createCmdResponse = dockerClient
-            .execCreateCmd(proxy.getContainerId())
-            .withCmd(
-                "sh",
-                "-c",
-                Stream.of(CouchbasePort.values())
-                    .map(port -> {
-                        return "/usr/bin/socat " +
-                            "TCP-LISTEN:" + port.getOriginalPort() + ",fork,reuseaddr " +
-                            "TCP:" + networkAlias + ":" + getMappedPort(port);
-                    })
-                    .collect(Collectors.joining(" & ", "true", ""))
-            )
-            .exec();
-
-        dockerClient.execStartCmd(createCmdResponse.getId())
-            .start()
-            .awaitCompletion(10, TimeUnit.SECONDS);
+    public final String getUsername() {
+        return username;
     }
 
-    @Override
-    public List<Integer> getExposedPorts() {
-        return proxy.getExposedPorts();
+    public final String getPassword() {
+        return password;
     }
 
-    @Override
-    public String getContainerIpAddress() {
-        return proxy.getContainerIpAddress();
+    public String getConnectionString() {
+        return String.format("couchbase://%s:%d", getContainerIpAddress(), getMappedPort(KV_PORT));
     }
 
     @Override
-    public Integer getMappedPort(int originalPort) {
-        return proxy.getMappedPort(originalPort);
-    }
+    protected void configure() {
+        super.configure();
+
+        WaitAllStrategy waitStrategy = new WaitAllStrategy();
+
+        // Makes sure that all nodes in the cluster are healthy.
+        waitStrategy = waitStrategy.withStrategy(
+            new HttpWaitStrategy()
+                .forPath("/pools/default")
+                .forPort(MGMT_PORT)
+                .withBasicCredentials(username, password)
+                .forStatusCode(200)
+                .forResponsePredicate(response -> {
+                    try {
+                        return Optional.of(MAPPER.readTree(response))
+                            .map(n -> n.at("/nodes/0/status"))
+                            .map(JsonNode::asText)
+                            .map("healthy"::equals)
+                            .orElse(false);
```

Comment:
private method ?
`boolean checkResponseForAllNodesHealthy(Response r)`

Reply:
private method inside lambda? :D I think the scope here is pretty small already